### ISSUE TITLE
[CP] Upgrading to Xcode 14.3 will not build when plugin transitive dependencies have a low deployment target

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#124529]
### issue_link

https://github.com/flutter/flutter/issues/124340

### Commit Hash

c60f20

### Target

stable

### pr_link

https://github.com/flutter/flutter/pull/124527

### Impacted Users

App developers running latest Xcode building iOS or macOS apps with certain plugins, including some firebase plugins

### Impact Description

When running latest Xcode (14.3) iOS and macOS apps will not build when using certain plugins that have dependencies with very low iOS target versions.  This includes any apps that use firebase_auth.

### Workaround

1. Downgrade to Xcode 14.2
2. Add a workaround to `ios/Podfile` and `macos/Podfile` that sets the pods to the same version the Flutter app is targeting.  https://stackoverflow.com/questions/75574268/missing-file-libarclite-iphoneos-a-xcode-14-3

<img width="546" alt="Screenshot 2023-04-06 at 1 23 41 PM" src="https://user-images.githubusercontent.com/682784/230486548-e82ad43a-9905-413d-a606-d2a2057da4a6.png">``

### Risk

low

### Test Coverage

yes

### Validation Steps

1. Check out 3.7.10 or be on master to pick up https://github.com/flutter/flutter/issues/123890
2. 
```
$ flutter create test_create
$ cd test_create
```
3. Add a plugin with a lower-than-9 iOS dependency target (I used `flutter_appauth`) as a pubspec dependency.
4.
```
$ flutter build ios
```

On master this does not build:
```
Error (Xcode): File not found: /Users/m/Applications/Xcode-14-3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/arc/libarclite_iphoneos.a


Error (Xcode): Linker command failed with exit code 1 (use -v to see invocation)
```
On this PR it does build.

**Comment by itsjustkevin** [comment]
Approving this as it has already been validated by @christopherfujino and @vashworth on the master channel.

**Comment by christopherfujino** [comment]
LGTM

**Comment by kenNg1** [comment]
Is there a workaround while we wait for this hotfix to come out?

**Comment by larssn** [comment]
> When running latest Xcode (14.2)

I think you mean 14.3 🙂

Anyway, setting the deployment target to 11 is not a valid workaround for us (in 14.3). It is already set to 13, and the error remains: `File not found: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/arc/libarclite_iphoneos.a`

This is just an FYI; we've looking at the workarounds suggested in https://github.com/flutter/flutter/issues/123890

**Comment by jmagman** [comment]
> I think you mean 14.3 🙂

Thanks, I did, updated.

> EDIT: This was the workaround that ended up working (replace the `post_install` lines in Podfile):

I linked to https://stackoverflow.com/questions/75574268/missing-file-libarclite-iphoneos-a-xcode-14-3 which says to set `IPHONEOS_DEPLOYMENT_TARGET` as you did.  The screenshot was to show you how to figure out what version to set it to.

We don't really need to document the workaround in detail in this hotfix request, you can check stackoverflow, my point is that the workaround involves manually updating files, which developers shouldn't need to do.

**Comment by jmagman** [comment]
@CaseyHillers since this was included in 3.7.11 (thank you!), can this be closed?
https://github.com/flutter/flutter/wiki/Hotfixes-to-the-Stable-Channel#3711-apr-12-2023
